test/cpp/end2end/client_callback_end2end_test.cc

```diff
@@ -423,51 +423,71 @@ TEST_P(ClientCallbackEnd2endTest, SimpleRpcExpectedError) {
 TEST_P(ClientCallbackEnd2endTest, SimpleRpcUnderLockNested) {
   MAYBE_SKIP_TEST;
   ResetStub();
-  std::mutex mu1, mu2, mu3;
-  std::condition_variable cv;
-  bool done = false;
+  struct Notifications {
```

Comment:
This whole thing is a bit awkward with the 1,2,3. What about a struct RPCParams or something and then passing around a vector of size 3 of them?